site/developer/developer_guide.md

```diff
@@ -0,0 +1,236 @@
+ 
+`"version": "[10,)"`
+  
+Specifies that an MBean attribute description is relevant for WebLogic Server version 10 and higher
+
+#### `wlst_mode`
+
+This key element specifies the WLST modes that are applicable for an MBean attribute description. The value can be "offline", "online", or "both".
+
+#### `wlst_name`
+
+This key element specifies the WLST name of the MBean attribute.
+
+#### `wlst_type`
+
+This key element specifies the data type used to set the WLST MBean attribute. Valid values are integer, long, string, boolean, jarray and MBean type. If the `wlst_read_type` is not set, this is also the data type used to read the value from WLST.
+
+#### `wlst_read_type`
+
+This key element specifies the data type used to read the WLST MBean attribute. If it is not specified, the value of `wlst_type` is used for the read.
+
+#### `get_method`
+
+This key element specifies which method should be used for retrieving the value the MBean attribute.  Valid values are: 
+
+- `GET`  use the WLST get method to retrieve the value of the attribute
+- `LSA`  use ls(type='a') to retrieve the value of the attribute
+- `NONE` do not retrieve the attribute value
+
+#### `preferred_model_type`
+
+This key element specifies the preferred data type that should be used to put data in the model during discovery. As an example, list values can be represented in the model as comma-separated text, such as `"value1, value2"`, or as a YAML list, such as `["value1", "value2"]`. If the list values can contain commas, it is preferred to use a YAML list.  
+
+#### `wlst_path`
+
+This key element specifies the name of the path expression used for navigating to the MBean attribute's folder. This name maps to an entry in the parent folder's `"wlst_paths": { }` list. 
+
+#### `value`
+
+This key element is used to specify the default value of the MBean attribute. For example:
+
+`"value": {"default": "${None:1}"}`
+
+#### `set_method`
+
+For cases where attributes cannot be set with simple types, it may be necessary to use a custom method to set the value. For example, most `Target` attributes require their values to be set as lists of MBeans, in online mode. This may be defined as: 
+
+```yaml
+    "attributes": {
+        "Target": [ {
+            "set_method": "MBEAN.set_target_mbeans", 
+            "set_mbean_type": "weblogic.management.configuration.TargetMBean"} ],
+    },
+```
+
+The method `set_target_mbeans` directs the tool to call the Jython method `attribute_setter.set_target_mbeans` to set this value.
+
+#### `set_mbean_type`
+
+When a `set_method` key is specified, it may be required to specify the MBean type for the set method to use (see the example under `set_method`).
+
+## Building WebLogic Deploy Tool
+
+### Prerequisite
+
+You will need the following software installed in your local build environment
+
+1. Oracle WebLogic Server installation version 10.3.6 and above
```

Comment:
Won't you need the WLS version to be 12.2.1.3 since the test compares the aliases against it?

Reply:
The unit test attributes_test.py is explicitly selecting 12.2.1.3 version of aliases and comparing names internally, so it doesn't care about underlying WLS version.
There may be other reasons for requiring a specific version, I think CER had a previous question about this.